### DISCUSSION TITLE
Binary executable for Debian OS

### DISCUSSION THREAD

**Discussion by ritam9495** [Z3Prover/z3#5232] [General]
I need the binary executable for Linux Debian OS. More specifically libz3.so would work as well as I guess I have all the other required files.

**Comment by aytey** [comment]
Have you tried https://github.com/Z3Prover/z3/releases/download/z3-4.8.10/z3-4.8.10-x64-ubuntu-18.04.zip? It says Ubuntu, but there's a good chance it is portable to other Linux's (e.g., it works for me on openSUSE Tumbleweed).

**Reply by aytey** [reply]
There's also https://github.com/Z3Prover/z3/releases/download/Nightly/z3-4.8.11-x64-glibc-2.31.zip if you want more bleeding edge (but 2.31 is more modern glibc vs. what might be in Ubuntu 18.04).

**Comment by ritam9495** [comment]
You can download one of the stable releases then follow the following commands:

`tar zxvf z3-4.8.10.tar.gz`
`cd z3-z3-4.8.10`
`python scripts/mk_make.py`
`cd build`
`make -j60 examples`

This will create all the necessary files and directories. To run the ./cpp_example:
either:
`setenv LD_LIBRARY_PATH ${LD_LIBRARY_PATH}:.`
then you can run ./cpp_example
or:
`env LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:. ./cpp_example` everytime

This will require no admin permission. Best for servers where you don't have any permissions (not even sudo)

**Reply by NikolajBjorner** [reply]
The builds use "ubuntu-latest" as specified by https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml

If 18.04 is more useful we can change the OS designation here:
https://github.com/Z3Prover/z3/blob/7869cdbbc88a70d7915f26c39259066985ea0d70/scripts/nightly.yaml#L25
